src/App.tsx

```diff
@@ -1,9 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState([...moviesFromServer]);
+
+  useEffect(() => {
+    setVisibleMovies(
+      searchQuery
+        ? moviesFromServer.filter(movie => {
+          const title = movie.title.toLowerCase();
+          const description = movie.description.toLowerCase();
+          const querry = searchQuery.trim().toLowerCase();
+
+          return title.search(querry) >= 0 || description.search(querry) >= 0;
+        })
+        : [...moviesFromServer],
+    );
+  }, [searchQuery]);
```

Comment:
I think we can get rid of this useEffect and useState. 
```suggestion
    const formattedQuery = query.toLowerCase().trim();

    const visibleMovies = moviesFromServer.filter(movie => {
          const title = movie.title.toLowerCase();
          const description = movie.description.toLowerCase();
          const querry = searchQuery.trim().toLowerCase();

          return title.search(querry) >= 0 || description.search(querry) >= 0;
  })
```